airbyte-cdk/python/unit_tests/sources/test_source_read.py

```diff
@@ -343,7 +343,7 @@ def test_concurrent_source_yields_the_same_messages_as_abstract_source_when_an_e
     source, concurrent_source = _init_sources([stream_slice_to_partition], state, logger)
     config = {}
     catalog = _create_configured_catalog(source._streams)
-    messages_from_abstract_source = _read_from_source(source, logger, config, catalog, state, RuntimeError)
+    messages_from_abstract_source = _read_from_source(source, logger, config, catalog, state, AirbyteTracedException)
```

Comment:
This is one area of deviation between the concurrent and regular abstract source. The behavior in `abstract_source.py` catches exceptions and emits trace errors. And at the end if at least one exception was found, it raises a final `AirbyteTracedException` that will make sure the process ends w/ a non-zero error code.

As far as I can tell, the concurrent just emits the same exception the source implementation emits. I am curious, but does that mean that a failure on one-stream in concurrent will derail all the concurrently running steams/slices?

I'm not entirely sure what parity should look like because the last error message for the regular abstract source is a summary of prior errors instead of a specific one. How do we feel about leaving this functionality outside of concurrent and filing a ticket to revisit this?

Reply:
your understanding is correct. Can we write a follow up issue for concurrent? I think it would be good to implement this as well